src/transformers/modeling_tf_utils.py

```diff
@@ -869,6 +869,18 @@ def compile(
             **kwargs,
         )
 
+    def compute_loss(self, *args, **kwargs):
+        if hasattr(tf.keras.Model, "compute_loss"):
+            # This will be true in TF 2.8 or greater
+            return super().compute_loss(*args, **kwargs)
+        else:
+            logger.warning(
+                "The old compute_loss method is deprecated as it conflicts with the Keras compute_loss "
+                "method added in TF 2.8. If you want the original HF compute_loss, please call "
+                "hf_compute_loss() instead."
+            )
```

Comment:
We use `warnings.warn(xxx, FutureWarning)` for deprecation warning.

Also mention it will be fully removed in v5 of Transformers.

Reply:
Done!